graylog2-server/src/main/java/org/graylog2/alerts/types/FieldValueAlertCondition.java

```diff
@@ -38,32 +43,85 @@
 import javax.annotation.Nullable;
 import java.text.DecimalFormat;
 import java.text.DecimalFormatSymbols;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Strings.isNullOrEmpty;
 
 public class FieldValueAlertCondition extends AbstractAlertCondition {
     private static final Logger LOG = LoggerFactory.getLogger(FieldValueAlertCondition.class);
 
-    public enum CheckType {
-        MEAN, MIN, MAX, SUM, STDDEV
+    enum CheckType {
+        MEAN("mean value"), MIN("min value"), MAX("max value"), SUM("sum"), STDDEV("standard deviation");
+
+        private final String description;
+
+        CheckType(String description) {
+            this.description = description;
+        }
+
+        public String getDescription() {
+            return description;
+        }
     }
 
-    public enum ThresholdType {
+    enum ThresholdType {
         LOWER, HIGHER
     }
 
     public interface Factory extends AlertCondition.Factory {
+        @Override
         FieldValueAlertCondition create(Stream stream,
                                         @Assisted("id") String id,
                                         DateTime createdAt,
                                         @Assisted("userid") String creatorUserId,
                                         Map<String, Object> parameters,
                                         @Assisted("title") @Nullable String title);
+        @Override
+        Config config();
+        @Override
+        Descriptor descriptor();
+    }
+
+    public static class Config implements AlertCondition.Config {
+        public Config() {
+        }
+
+        @Override
+        public ConfigurationRequest getRequestedConfiguration() {
+            return ConfigurationRequest.createWithFields(
+                new TextField("field", "Field", "", "Field name that should be checked", ConfigurationField.Optional.NOT_OPTIONAL),
+                new NumberField("time", "Time Range", 0, "Time span in seconds to check", ConfigurationField.Optional.NOT_OPTIONAL),
+                new NumberField("threshold", "Threshold", 0.0, "Value which triggers an alert if crossed", ConfigurationField.Optional.NOT_OPTIONAL),
+                new DropdownField(
+                    "threshold_type",
+                    "Threshold Type",
+                    ThresholdType.HIGHER.toString(),
+                    Arrays.stream(ThresholdType.values()).collect(Collectors.toMap(Enum::toString, thresholdType -> thresholdType.toString().toLowerCase(Locale.ENGLISH))),
```

Comment:
Since this pattern is used quite often (also in other entities), we should extract it into a common method.

Suggestion:
```java
public static Map<String, String> buildEnumMap(Class<? extends Enum> enumClass, Function<Enum, String> valueMapping) {
    return Arrays.stream(enumClass.getEnumConstants()).collect(Collectors.toMap(Enum::toString, valueMapping));
}

enum TestEnum {
    ONE, TWO
}

@Test
public void testBuildEnumMap() throws Exception {
    final Map<String, String> enumMap = buildEnumMap(TestEnum.class, (t) -> t.name().toLowerCase());
    assertThat(enumMap)
            .containsEntry("ONE", "one")
            .containsEntry("TWO", "two");
}
```